components/layout_2020/display_list/stacking_context.rs

```diff
@@ -237,24 +283,28 @@ pub struct StackingContext {
     /// Fragments that make up the content of this stacking context.
     fragments: Vec<StackingContextFragment>,
 
-    /// All non-float stacking context and pseudo stacking context children
-    /// of this stacking context.
-    stacking_contexts: Vec<StackingContext>,
-
-    /// All float stacking container children of this stacking context.
-    /// These are stored separately because they should not be passed up to
-    /// their real stacking context ancestors. From the definition of stacking
-    /// containers from <https://drafts.csswg.org/css-position-4#painting-order>:
-    ///
+    /// Children that need to be stolen by the parent stacking context if this
+    /// is a stacking container, that is, real stacking contexts and positioned
+    /// stacking containers (where ‘z-index’ is auto).
+    /// <https://drafts.csswg.org/css-position-4#painting-order>
     /// > To paint a stacking container, given a box root and a canvas canvas:
     /// >  1. Paint a stacking context given root and canvas, treating root as
     /// >     if it created a new stacking context, but omitting any positioned
     /// >     descendants or descendants that actually create a stacking context
     /// >     (letting the parent stacking context paint them, instead).
-    ///
-    /// Note that all stacking containers / pseudo stacking contexts are passed up
-    /// to parent stacking contexts, except in the case of floats.
-    float_stacking_contexts: Vec<StackingContext>,
+    stealable_children: Vec<StackingContext>,
+
+    /// Float stacking containers.
+    /// These are separate from [Self::stealable_children] because they should
+    /// never be stolen by the parent stacking context.
+    float_stacking_containers: Vec<StackingContext>,
+
+    /// Atomic inline stacking containers.
+    /// These are separate from [Self::stealable_children] because they should
+    /// never be stolen by the parent stacking context, and separate from
+    /// [Self::float_stacking_containers] so that [StackingContextFragment] can
+    /// index into this vec to paint them in fragment order.
```

Comment:
Might be good to put a pointer to the relevant part of the specification here.

Reply:
done in 8e60b20ca11c2a3d755e21931661c14d108f9261